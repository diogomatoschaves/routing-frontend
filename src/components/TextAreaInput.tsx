import React from 'react'
import { Form, TextArea } from 'semantic-ui-react'
import styled from 'styled-components'
import { HandleValueUpdate, InputValues, UpdateState } from '../types'
import { PETROL_3 } from '../utils/colours'

interface Props {
  id: string
  handleValueUpdate?: HandleValueUpdate
  handleInputChange?: UpdateState
  rows: number
  color?: string
  editableValue: string
  placeholder?: string
  setInputRef?: (ref: any) => void
  inputValues: InputValues
}

const StyledForm = styled(Form)`
  width: 100%;

  &.ui.form textarea {
    width: 100%;
    font-size: 16px;
    border-radius: 7px;
    border: none;
    color: ${props => (props.color ? props.color : 'rgb(100, 100, 100)')};
    background-color: rgb(250, 250, 250);
    user-select: text;
    z-index: 1000;
    font-family: 'BasisGrotesque Light', Lato, 'Helvetica Neue', Arial, Helvetica,
      sans-serif !important;

    &:focus {
      background-color: rgb(248, 248, 248);
      color: ${props => (props.focuscolor ? props.focuscolor : 'black')};
    }

    &::-webkit-scrollbar {
      width: 8px;
    }

    &::-webkit-scrollbar-track {
      box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
      border-radius: 10px;
    }

    &::-webkit-scrollbar-thumb {
      border-radius: 10px;
      box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.5);
    }
  }
`

export default function TextAreaInput({
  id,
  handleValueUpdate,
  handleInputChange,
  rows,
  color,
  editableValue,
  placeholder,
  setInputRef,
  inputValues
}: Props) {
  return (
    <StyledForm color={color} focuscolor={PETROL_3}>
      <TextArea
        ref={input => setInputRef && setInputRef(input)}
        id={id}
        className={'textarea'}
        placeholder={placeholder ? placeholder : ''}
        rows={rows}
        value={editableValue}
        onChange={(e, { id: newId, value }) =>
          handleInputChange &&
          handleInputChange('inputValues', {
            ...inputValues,
            [newId]: value
          })
        }
        onFocus={(e: any) => !handleInputChange && e.target.select()}
        onBlur={() =>
          handleValueUpdate && handleValueUpdate({ id, value: editableValue })
        }
      />
    </StyledForm>
  )
}
