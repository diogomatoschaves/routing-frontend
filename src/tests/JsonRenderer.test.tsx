import React from 'react'
import TestRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import TextAreaInput from '../components/TextAreaInput';

const mockEventStart = { lng: 13.389869, lat: 52.510348 }
const mockEventEnd = { lng: 13.39114, lat: 52.510425 }

const mockBody = {
  locations: [
    { 
      lat: mockEventStart.lat,
      lon: mockEventStart.lng
    },
    { 
      lat: mockEventEnd.lat,
      lon: mockEventEnd.lng
    },
  ],
  reportGeometry: true,
}


it('expect to render TextAreaInput component', () => {

  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <TextAreaInput 
        value={mockBody}
        editableValue={JSON.stringify(mockBody)}
        id={'bodyValue'}
        editJson={false}
        handleValueUpdate={jest.fn()}
        handleInputChange={jest.fn()}
        rows={10}
        color={'rgb(100, 100, 100)'}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
}) 
