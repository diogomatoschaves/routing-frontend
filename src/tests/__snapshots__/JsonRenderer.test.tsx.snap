// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render JsonRenderer component 1`] = `
<form
  className="ui form sc-ifAKCX cFwNVD"
  color="rgb(100, 100, 100)"
  focuscolor="#5097ab"
  onSubmit={[Function]}
>
  <div
    className="sc-bwzfXH pbVzJ"
  >
    <div
      className="sc-bdVaJa Pwmcq"
      onClick={[Function]}
    >
      
      <i
        aria-hidden="true"
        className="caret up icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="sc-bwzfXH pbVzJ"
    >
      <div
        className="sc-bdVaJa Pwmcq"
        onClick={[Function]}
      >
        Locations
        <i
          aria-hidden="true"
          className="caret down icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="sc-bwzfXH hGrrjl"
    >
      <span
        className="sc-htpNat hCIKUk"
      >
        Report Geometry
        : 
      </span>
      <span
        className="sc-bxivhb bKAtgj"
      >
        true
      </span>
    </div>
  </div>
</form>
`;
